src/app/features/tech-record/create/components/batch-create/batch-create.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GlobalError } from '@core/components/global-error/global-error.interface';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { FormNodeWidth } from '@forms/services/dynamic-form.types';
+import { CustomValidators } from '@forms/validators/custom-validators';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { combineLatest, debounceTime, Subject, take, takeUntil, withLatestFrom } from 'rxjs';
+@Component({
+  selector: 'app-batch-create',
+  templateUrl: './batch-create.component.html'
+})
+export class BatchCreateComponent implements OnInit, OnDestroy {
+  readonly maxNumberOfVehicles = 39;
+  private destroy$ = new Subject<void>();
+  form: FormGroup;
+
+  constructor(
+    private fb: FormBuilder,
+    private globalErrorService: GlobalErrorService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private technicalRecordService: TechnicalRecordService
+  ) {
+    this.form = this.fb.group({
+      numberOfVehicles: [0, [Validators.max(39)]],
+      vehicles: this.fb.array([])
+    });
+  }
+
+  private startingObject = { vin: '' };
+
+  ngOnInit() {
+    this.technicalRecordService.editableVehicleTechRecord$.pipe(take(1)).subscribe(vehicle => {
+      if (!vehicle) {
+        this.back();
+      }
+    });
+
+    combineLatest([this.technicalRecordService.batchVehicles$, this.technicalRecordService.batchCount$, this.technicalRecordService.generateNumber$])
+      .pipe(take(1))
+      .subscribe({
+        next: ([vehicles, numberOfVehicles, generateNumber]) => {
+          if (numberOfVehicles) {
+            this.addVehicles(numberOfVehicles, generateNumber);
+            this.form.setValue({ numberOfVehicles, vehicles });
+          }
+        }
+      });
+
+    this.watchNumberOfVehicles();
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  watchNumberOfVehicles() {
+    this.form
+      .get('numberOfVehicles')
+      ?.valueChanges.pipe(withLatestFrom(this.technicalRecordService.generateNumber$.pipe(take(1))), takeUntil(this.destroy$), debounceTime(500))
+      .subscribe({
+        next: ([val, generateNumber]) => {
+          const n = Math.min(val, this.maxNumberOfVehicles);
+          if (isNaN(n) || n > this.maxNumberOfVehicles) {
+            return;
+          }
+
+          const currentList = this.vehicles.controls.length;
+
+          if (currentList > val) {
+            this.vehicles.controls.length = n;
+          } else {
+            this.addVehicles(n - currentList, generateNumber);
+          }
+        }
```

Comment:
The `Math.min()` is not needed here.
```suggestion
          if (isNaN(val) || val > this.maxNumberOfVehicles) {
            return;
          }

          if (this.vehicles.controls.length > val) {
            this.vehicles.controls.length = val;
          } else {
            this.addVehicles(val - this.vehicles.controls.length, generateNumber);
          }
```